strawberry/types/federation/object_type.py

```diff
@@ -0,0 +1,79 @@
+from typing import List, Optional, Union
+
+from strawberry.types import StrawberryObjectType, StrawberryUnion
+
+_OBJECT_FED_TYPE = Union[StrawberryObjectType, 'StrawberryObjectTypeFederation']
+
+
+# TODO: Should federation use composition or inheritance?
+#       i.e. should StrawberryObjectTypeFederation inherit from
+#       StrawberryObjectType or just hold a reference to one?
+
+# TODO: This name is probably too long and too verbose. Think of something
+class StrawberryObjectTypeFederation:
+    """
+
+    TODO: One decorator for federation? Or multiple? This feels a bit stacked
+    >>> @strawberry.key("upc")
+    ... @strawberry.extends(OtherType)
```

Comment:
nit; extends doesn’t extend another type, it extends the current type in the federated GraphQL (based on the name) so it doesn’t need to accept a param :)